applications/minotari_app_utilities/src/network_check.rs

```diff
@@ -72,10 +73,42 @@ pub fn is_network_choice_valid(network: Network) -> Result<Network, NetworkCheck
 pub fn set_network_if_choice_valid(network: Network) -> Result<(), NetworkCheckError> {
     match is_network_choice_valid(network) {
         Ok(network) => {
-            let mut current_network = CURRENT_NETWORK.lock()?;
-            *current_network = network;
-            Ok(())
+            // let mut current_network = Network::current();
+            Network::set_current(network).map_err(|instead_network| NetworkCheckError::CouldNotSetNetwork {
+                attempted: network,
+                current_network: instead_network,
+            })
         },
         Err(e) => Err(e),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // It's not clear how to test this without changing the global state
+    // I considered adding a test for `is_network_choice_valid` but that would be redundant
+    // since the function is trivial and is only used by `set_network_if_choice_valid`
+    #[ignore = "This test changes the global network state and can affect other tests"]
+    #[test]
+    fn set_network() {
+        assert_eq!(Network::current(), Network::Esmeralda);
+
+        // Invalid networks
+        for network in [Network::MainNet, Network::StageNet, Network::NextNet] {
+            assert!(set_network_if_choice_valid(network).is_err());
+            assert_eq!(Network::current(), Network::Esmeralda);
+        }
+
+        // Valid networks (we can't test against other binary configurations)
+        for network in [Network::LocalNet, Network::Igor, Network::Esmeralda] {
+            assert!(set_network_if_choice_valid(network).is_ok());
```

Comment:
This fails for me locally on the first trip through the loop. The issue is that `Network::set_current` returns an error.